client/core/core_test.go

```diff
@@ -2675,3 +2675,69 @@ func TestHandleMatchProofMsg(t *testing.T) {
 		t.Fatalf("[handleMatchProofMsg] unexpected error: %v", err)
 	}
 }
+
+func TestAddrHost(t *testing.T) {
+	tests := []struct {
+		name, addr, want string
+		wantErr          bool
+	}{{
+		name: "scheme, host, and port",
+		addr: "https://localhost:5758",
+		want: "localhost:5758",
+	}, {
+		name: "host and port",
+		addr: "localhost:5758",
+		want: "localhost:5758",
+	}, {
+		name: "just port",
+		addr: ":5758",
+		want: "localhost:5758",
+	}, {
+		name: "ip host and port",
+		addr: "127.0.0.1:5758",
+		want: "127.0.0.1:5758",
```

Comment:
Do want this to parse to localhost too?

Reply:
No, the IP is valid and the user has a reason presumably.